chain/tests/chain_test_helper.rs

```diff
@@ -11,6 +11,7 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+#![allow(dead_code)]
```

Comment:
What is that needed for?

Reply:
there's some cargo oddness that's causing a 'not used' warning for the `clean_output_dir` function below during compilation, even though it's configured as test only. I spent some time trying to track it down and couldn't find a reason for it. I could possibly spend more time on it

Reply:
Maybe ask around on IRC #rust or other rust forums if other people have run into that issue. I see that googling for
   rust 'not used' warning for test code
also brings up discussions  on stack overflow, reddit, and the Rust programming language forum...